osgar/drivers/rosproxy.py

```diff
@@ -0,0 +1,159 @@
+"""
+  ROS (Robot Operating System) Proxy
+"""
+
+from threading import Thread
+import struct
+from xmlrpc.client import ServerProxy
+from xmlrpc.server import SimpleXMLRPCServer
+import math
+
+import socket
+
+
+from osgar.bus import BusShutdownException
+
+NODE_HOST, NODE_PORT = ('127.0.0.1', 8000)
+PUBLISH_PORT = 8123
+
+
+ROS_MESSAGE_TYPES = {
+    'std_msgs/String': '992ce8a1687cec8c8bd883ec73ca41d1',
+    'geometry_msgs/Twist': '9f195f881246fdfa2798d1d3eebca84a',
+    'std_msgs/Imu': '6a62c6daae103f4ff57a132d6f95cec2',
+
+    'sensor_msgs/LaserScan': '90c7ef2dc6895d81024acba2ac42f369',
+    'sensor_msgs/Image': '060021388200f6f0f447d0fcd9c64743',
+    'sensor_msgs/CompressedImage': '8f7a12909da2c9d3332d540a0977563f',
+    'nav_msgs/Odometry': 'cd5e73d190d741a2f92e81eda573aca7',
+}
+
+
+def prefix4BytesLen(s):
+    "adding ROS length"
+    if type(s) == str:
```

Comment:
Why would anyone give unicode string to this function? Or maybe I don't understand the intent of the function.

Reply:
This was due to port from Python2, when original "hello world" was sending string. I will check it, thanks.